### ISSUE TITLE
Tracking does not work when using context.Update()

### ISSUE THREAD

**Issue by thoraj** [gnaeus/EntityFramework.CommonTools#4]
EF Core contexts support an Update method which will update an entity if it exist. 
When using context.Update(), the columns for tracking CreatorUserId and CreatedUTC are reset. 

Is this a known problem? 
If so, is this something that is on the roadmap to fix?

**Comment by gnaeus** [comment]
@thoraj, can you give an example of your scenario? I don't understand what do you mean.

Suppose we have an entity `Post` with `CreatorUserId` and `CreatedUtc` properties.
If we call `context.Update(post)`, the following SQL will be generated during `SaveChanges`:
```sql
UPDATE Posts
SET CreatorUserId = @p0,
    CreatedUtc = @p1
    ...
WHERE Id = @p2;
```
But we get same SQL if we manually owerwrite `post.CreatedUtc = new DateTime(2018, 1, 1);`

If we load `Post` from database, the `CreatedUtc` property already contains a date. Then `context.Update` marks all properties as changed. And UPDATE statement doesn't really change it's value.

`CreatedUtc` and `CreatorUserId` will be erased in DB only if we have detached entity with empty `CreatedUtc` and `CreatorUserId` fields. And then we call `context.Update()` on it. Is this your scenario?

**Comment by gnaeus** [comment]
You are right. Assigning `DateTime.Zero` to `CreatedUtc` and  `default(TUserId)` to `CreatorUserId` is meaningless. But if we explicitely change `CreatedUtc` to non-default value, I think we should change it in DB too.

I can add this behaviour in the next release.

**Comment by gnaeus** [comment]
Fixed in [2.0.2](https://github.com/gnaeus/EntityFramework.CommonTools/releases/tag/2.0.2)
